src/pages/cli/graphql/data-modeling.mdx

```diff
@@ -549,7 +549,494 @@ type Address @model {
+  projectTeamTeamId: ID # auto generated child primary key reference
+  projectTeamName: String # auto generated child sort key reference
+}
+
+input UpdateProjectInput {
+  projectId: ID!
+  name: String!
+  projectTeamTeamId: ID # auto generated child primary key reference
+  projectTeamName: String # auto generated child sort key reference
+}
+```
+</Block>
+
+<Block name='Implicit Bi-directional "has many"'>
+
+```graphql
+type Post @model {
+  postId: ID! @primaryKey(sortKeyFields:["title"])
+  title: String!
+  comments: [Comment] @hasMany
+}
+
+type Comment @model {
+  commentId: ID! @primaryKey(sortKeyFields:["content"])
+  content: String!
+  post: Post @belongsTo
+}
+```
+This generates additional foreign key fields only in child type.
+
+Model fields:
+```graphql
+type Comment {
+  commentId: ID!
+  content: String!
+  post: Post
+  createdAt: AWSDateTime!
+  updatedAt: AWSDateTime!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+```
+Input fields:
+```graphql
+input CreateCommentInput {
+  commentId: ID!
+  content: String!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+
+input UpdateCommentInput {
+  commentId: ID!
+  content: String!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+```
+</Block>
+
+<Block name='Implicit Uni-directional "has many"'>
+
+```graphql
+type Post @model {
+  postId: ID! @primaryKey(sortKeyFields:["title"])
+  title: String!
+  comments: [Comment] @hasMany
+}
+
+type Comment @model {
+  commentId: ID! @primaryKey(sortKeyFields:["content"])
+  content: String!
+}
+```
+This generates additional foreign key fields only in child type.
+
+Model fields:
+```graphql
+type Comment {
+  commentId: ID!
+  content: String!
+  createdAt: AWSDateTime!
+  updatedAt: AWSDateTime!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+```
+Input fields:
+```graphql
+input CreateCommentInput {
+  commentId: ID!
+  content: String!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+
+input UpdateCommentInput {
+  commentId: ID!
+  content: String!
+  postCommentsPostId: ID # auto generated parent primary key reference
+  postCommentsTitle: String # auto generated parent sort key reference
+}
+```
+</Block>
+
+</BlockSwitcher>
+
+You can also customize the foreign key fields instead of the auto-generated ones in the schema by **explicitly** defining the `fields` parameter in relation directives.
+
+<Callout>
+The order in "fields" parameter matters. The first element should refer the primary key field and the rest are sort key fields.
```

Comment:
```suggestion
The order of the elements in the value of "fields" argument matters. The first element should always be the primary key of the associated model; and remaining elements should match the `sortKeyFields` specified in the `@primaryKey` directive of t he associated model.
```

Reply:
This is true for `hasOne` and `belongsTo`. But for `hasMany` it is different. When defining `@hasMany(fields:)`, We are passing the PK and SK fields from the parent and associate it with the child secondary index. So I think the `associated model` here is not accurate enough to cover all cases.

Reply:
I temporarily replace the word with `relational`